src/snowflake/cli/plugins/snowpark/commands.py

```diff
@@ -164,18 +170,22 @@ def deploy(
         raise ClickException(msg)
 
     # Create stage
-    stage_name = snowpark.stage_name
-    stage_manager = StageManager()
-    stage_name = FQN.from_string(stage_name).using_context()
-    stage_manager.create(
-        stage_name=stage_name, comment="deployments managed by Snowflake CLI"
-    )
-
     snowflake_dependencies = _read_snowflake_requrements_file(
         paths.snowflake_requirements_file
     )
-
-    artifact_stage_directory = get_app_stage_path(stage_name, snowpark.project_name)
+    stage_names = {
+        entity.stage for entity in [*functions.values(), *procedures.values()]
+    }
+    stage_manager = StageManager()
+    project_name = (
+        pd.defaults.project_name if pd.defaults.project_name else "my_snowpark_project"
+    )
+    for stage in stage_names:
+        stage = FQN.from_string(stage).using_context()
+        stage_manager.create(
+            stage_name=stage, comment="deployments managed by Snowflake CLI"
+        )
+        artifact_stage_directory = get_app_stage_path(stage, project_name)
```

Comment:
We create multiple stages but use only one `artifact_stage_directory`? I think we may want to consider doing
```
for stage in stages:
   create_stage(stage)
   for udf_sproc in stage_to_objects_map[stage]:
       _deploy_single_object()
```

WDYT? In this way we are processing stage by stage. In case of single stage there's no difference compared to current bahavior.

Reply:
In what way? Currently new `artifact_stage_directory` is created for each stage. 
We use common project name, as we can have multiple udfs on single stage, and need a name to group them.

Reply:
This loop:
```
for stage in stage_names:
        stage = FQN.from_string(stage).using_context()
        stage_manager.create(
            stage_name=stage, comment="deployments managed by Snowflake CLI"
        )
        artifact_stage_directory = get_app_stage_path(stage, project_name)
```

Results in `artifact_stage_directory` being directory for last stage in `stage_names`. And this single value is used for all entities independent of their stages.

Reply:
Corrected - now it's all in a loop.